src/typ.rs

```diff
@@ -1960,6 +1969,17 @@ fn infer_seq(
 ) -> Result<TypedExpr, Error> {
     let first = infer(first, level, env)?;
     let then = infer(then, level, env)?;
+
+    match &*first.typ() {
```

Comment:
Same style thing again, can we swap `&*` for `as_ref()`? Thank you

Reply:
Do you want me to change this line too (in another commit, of course): https://github.com/gleam-lang/gleam/blob/master/src/typ.rs#L2093

Reply:
Yes please!

Reply:
It actually looks like the `AsRef` trait isn't implemented on `TypedExpr`. Do you want me to implement it and then use `as_ref()`?

Reply:
Sorry I've been unclear. It's implemented on `Arc` (a fancy pointer type) which `typ()` returns so `first.typ().as_ref()`

Reply:
Ah, that makes more sense! Thanks, I pushed up the fix.